pybigquery/sqlalchemy_bigquery.py

```diff
@@ -367,17 +369,12 @@ def _get_table(self, connection, table_name, schema=None):
         if isinstance(connection, Engine):
             connection = connection.connect()
 
-        table_name_prepared = dataset = project = None
-        if self.dataset_id:
-            table_name_prepared = table_name
-            dataset = self.dataset_id
-            project = None
-
-        else:
-            project, dataset, table_name_prepared = self._split_table_name(table_name)
-            if dataset is None and schema is not None:
-                table_name_prepared = table_name
+        project, dataset, table_name_prepared = self._split_table_name(table_name)
+        if dataset is None:
+            if schema is not None:
```

Comment:
Is precedence for the 'table name dataset' over the scheme the behavior in other sqlalchemy dialects?

Reply:
Unlike BigQuery, Postgres has a two level hierarchy (schema and table).

Postgres doesn't seem to do any interpretation of schema from the tablename parameter: https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/dialects/postgresql/base.py#L2599-L2635

Same situation with MySQL:

https://github.com/zzzeek/sqlalchemy/blob/master/lib/sqlalchemy/dialects/mysql/base.py#L2458-L2491

Arguably BigQuery is a special situation given the three level hierarchy (project/dataset/table).  Other common databases let you run different database instances on the same server and with the same binaries, but BigQuery allows all relevant project_ids to be reached via the same connection and joined with each other, so it's more like a two-level schema name than anything else.

Reply:
Makes sense.
What do you think should be the behavior for BQ?
Raising an error in that case sounds legit to me, problem is we will be changing a fundamental behavior.
Anyway this should probably be documented.

Reply:
I mean, ideally I think the parsing on the table name should be dropped entirely, and we should only parse the schema.  Given you already parse on the table name, it's probably rude to stop doing so, as there are probably code bases that rely on it.

That said, if we maintain compatibility there, I do think raising an exception when the user gives you conflicting instructions is safer than arbitrarily choosing which one to obey, and wouldn't break any sane existing code.